plugins/utils/which-key.nix

```diff
@@ -559,26 +559,72 @@ lib.nixvim.neovim-plugin.mkNeovimPlugin config {
       '';
       visible = false;
     };
+
+    mappingsWithOpts = mkOption {
+      type =
+        with types;
+        listOf (submodule {
+          options = {
+            spec = mkOption {
+              type = listOf (attrsOf anything);
+              description = ''
+                WhichKey mappings following the same setup as `plugins.which-key.settings.spec`.
+              '';
+            };
+            opts = mkOption {
+              type = attrsOf anything;
+              description = ''
+                WhichKey mapping options to apply to all the mapping in corresponding `plugins.which-key.mappingsWithOpts`.
+              '';
+            };
+          };
+        });
```

Comment:
I think we can set `default = [ ]` here instead of checking `isDefined` below in our usage.

Since it isn't a deprecated option we don't need to avoid having a definition, and having a default will make it easier for end-users to depend on the option value in any of their config.